### ISSUE TITLE
Command failed for target `gmysqlbackend.lo'

### ISSUE THREAD

**Issue by HLFH** [PowerDNS/pdns#3849]
**PowerDNS version:** tag auth-4.0.0-alpha3

**Where you got the software from (e.g. distribution, compiled yourself):** git clone --branch auth-4.0.0-alpha3 https://github.com/PowerDNS/pdns.git

**Operating System and version:** ![capture d ecran 2016-05-11 a 14 53 56](https://cloud.githubusercontent.com/assets/2359059/15181571/3995f00e-1788-11e6-899d-500a9ca4235e.png)

**Steps to reproduce: How can we reproduce the issue:**
```bash
sudo -i
git clone --branch auth-4.0.0-alpha3 https://github.com/PowerDNS/pdns.git
cd pdns/
git describe --tags
./bootstrap
./configure --prefix=/usr --libexecdir='$${prefix}/lib' --libdir='$${prefix}/lib/powerdns' --sysconfdir=/etc/powerdns --infodir='$${datadir}/info' --mandir='$${datadir}/man' --with-modules="gmysql geoip"
make
```
**Expected behavior: what did you expect what would happen?**: I expected a correct compilation of  auth-4.0.0-alpha3.

**Observed behavior: what actually happend when following the steps?**: During gmysqlbackend.lo compilation, we get the issue _Command failed for target `gmysqlbackend.lo'_
```bash
[root@arch-server pdns]# make
make  all-recursive
make[1] : on entre dans le répertoire « /root/pdns »
Making all in ext
make[2] : on entre dans le répertoire « /root/pdns/ext »
Making all in yahttp
make[3] : on entre dans le répertoire « /root/pdns/ext/yahttp »
Making all in yahttp
make[4] : on entre dans le répertoire « /root/pdns/ext/yahttp/yahttp »
make[4]: rien à faire pour « all ».
make[4] : on quitte le répertoire « /root/pdns/ext/yahttp/yahttp »
make[4] : on entre dans le répertoire « /root/pdns/ext/yahttp »
make[4]: rien à faire pour « all-am ».
make[4] : on quitte le répertoire « /root/pdns/ext/yahttp »
make[3] : on quitte le répertoire « /root/pdns/ext/yahttp »
Making all in json11
make[3] : on entre dans le répertoire « /root/pdns/ext/json11 »
make[3]: rien à faire pour « all ».
make[3] : on quitte le répertoire « /root/pdns/ext/json11 »
make[3] : on entre dans le répertoire « /root/pdns/ext »
make[3]: rien à faire pour « all-am ».
make[3] : on quitte le répertoire « /root/pdns/ext »
make[2] : on quitte le répertoire « /root/pdns/ext »
Making all in modules
make[2] : on entre dans le répertoire « /root/pdns/modules »
Making all in gmysqlbackend
make[3] : on entre dans le répertoire « /root/pdns/modules/gmysqlbackend »
  CXX      gmysqlbackend.lo
g++: erreur interne du compilateur: Processus arrêté (program cc1plus)
Veuillez soumettre un rapport d’anomalies complet,
avec la sortie du préprocesseur si nécessaire.
Voir <https://bugs.archlinux.org/> pour les instructions.
Makefile:599 : la recette pour la cible « gmysqlbackend.lo » a échouée
make[3]: *** [gmysqlbackend.lo] Erreur 1
make[3] : on quitte le répertoire « /root/pdns/modules/gmysqlbackend »
Makefile:502 : la recette pour la cible « all-recursive » a échouée
make[2]: *** [all-recursive] Erreur 1
make[2] : on quitte le répertoire « /root/pdns/modules »
Makefile:551 : la recette pour la cible « all-recursive » a échouée
make[1]: *** [all-recursive] Erreur 1
make[1] : on quitte le répertoire « /root/pdns »
Makefile:483 : la recette pour la cible « all » a échouée
make: *** [all] Erreur 2
```
**Relevant logs:** where could I find them? Is this really needed?
FYI, I'm using https://www.archlinux.org/packages/extra/x86_64/mariadb/ and https://www.archlinux.org/packages/extra/i686/libmariadbclient/. Is that an issue it's a little out-of-date? 10.1.14 is now available. Or is PowerDNS only compatible with MySQL and no longer with MariaDB?

**Comment by HLFH** [comment]
IMHO related to https://github.com/PowerDNS/pdns/blob/auth-4.0.0-alpha3/modules/gmysqlbackend/gmysqlbackend.cc & https://github.com/PowerDNS/pdns/blob/auth-4.0.0-alpha3/modules/gmysqlbackend/smysql.cc & https://github.com/PowerDNS/pdns/blob/auth-4.0.0-alpha3/modules/gmysqlbackend/smysql.hh

**Comment by pieterlexis** [comment]
Hello @HLFH,

Thank you for your your extended bug report.

In my experience, `g++: erreur interne du compilateur: Processus arrêté (program cc1plus)` (the english variant of it) means that either the compiler is broken (which is most likely not the case) or the compiler process is stopped. I see that your machine has 512mb of RAM, this is not enough to build PowerDNS. Could you add 2GB of swapspace or try building on a machine with 2GB of RAM?

PowerDNS with the MySQL backend builds for me on my Arch machine with GCC 6.1.1.20160501 and libmariadbclient 10.1.13-1

**Comment by HLFH** [comment]
@pieterlexis Ok. Thanks. I'll try a droplet with more RAM (or with swapspace) and I come back to you.

**Comment by HLFH** [comment]
@pieterlexis Solved thanks! First time of my life that swap is useful to me!! I followed this [tutorial](https://www.digitalocean.com/community/tutorials/how-to-add-swap-space-on-ubuntu-16-04) until the *Enabling the Swap File* chapter. It seems that we should not use some swap partition or file [on SSDs](http://askubuntu.com/questions/652337/why-no-swap-partitions-on-ssd-drives) because it could harm them. So, I created a swap file of 2GB for the `pdns` compilation, and I removed it after with `swapoff -a` and `rm -f /swapfile`.
